asdf/asdf.py

```diff
@@ -1757,11 +1764,18 @@ def is_asdf_file(fd):
     For URL input, looks for an extension that should be passed
     to AsdfInFits, otherwise assumes ASDF.
 
+    This method is deprecated in favor of `~asdf.util.get_file_type`.
+
     Parameters
     ----------
     fd : str, `~asdf.generic_io.GenericFile`
 
     """
+    warnings.warn(
+        "The asdf.asdf.is_asdf_file method is deprecated.  Use asdf.util.get_file_type instead.",
+        AsdfDeprecationWarning,
+    )
```

Comment:
The `is_asdf_file()` method is only used internally, right?  Or is it used elsewhere?

Reply:
I'm not sure.  It's tested as though it were part of the public API, but it's hard to tell since historically we haven't been careful about clearly marking public vs internal.  GitHub search only returns hits for the asdf repo for the string "is_asdf_file".

Reply:
I'll remove it...